common/rs-config.h

```diff
@@ -87,6 +87,105 @@ namespace rs2
 
         static config_file& instance();
 
+        // Retrieves a value from a nested JSON structure using dot notation
+        template< typename T >
+        T get_nested( const std::string & path ) const
+        {
+            std::istringstream ss( path );
+            std::string token;
+            const rsutils::json * current = &_j;
+
+            while( std::getline( ss, token, '.' ) )
+            {
+                if( ! current->contains( token ) )
+                {
+                    return config_value( "" );
+                }
+                current = &( *current )[token];  // getting to the next level in the JSON structure
+            }
+            return current->get< T >();
+        }
+
+        // Sets a value in a nested JSON structure using dot notation
+        template< typename T >
+        void set_nested( const std::string & path, const T & val )
+        {
+            std::vector< std::string > keys;
+            std::istringstream ss( path );
+            std::string token;
+
+            while( std::getline( ss, token, '.' ) )
+            {
+                keys.push_back( token );
+            }
+            rsutils::json * current = &_j;
+
+            for( size_t i = 0; i < keys.size() - 1; ++i )
+            {
+                if( ! current->contains( keys[i] ) )
+                {
+                    ( *current )[keys[i]] = rsutils::json::object();
+                }
+                current = &( *current )[keys[i]];
```

Comment:
I think that working with a reference, not a pointer, will be easier and the syntax clearer